Plugins/GitUIPluginInterfaces/BuildServerIntegration/BuildInfo.cs

```diff
@@ -16,10 +16,26 @@ public enum BuildStatus
 
         public string Id { get; set; }
         public DateTime StartDate { get; set; }
+        public long? Duration { get; set; }
         public BuildStatus Status { get; set; }
         public string Description { get; set; }
         public string[] CommitHashList { get; set; }
         public string Url { get; set; }
         public bool ShowInBuildReportTab { get; set; } = true;
+
+        public string DurationText()
```

Comment:
This overloads the DTO with behaviors, which isn't good.
I don't see any reason for this to be a method, it renders the set state, so it is deterministic.
Perhaps it is better to:
* rename `string DurationText()` to `string DurationFormatted { get; }`
* change the `Duration` property to set `DurationFormatted` upon assignment

Reply:
>change the Duration property to set DurationFormatted upon assignment

I need the raw value to be able to compute the end date.

>rename string DurationText() to string DurationFormatted { get; }

It looks odd to me to pretend it is a property. The same way toString() is a function not a property (being deterministic as well). I want to provide a common, consistent way across all adapters to format the duration.

Reply:
I meant the following:

```
long? Duration 
{ 
    get { return _duration; }
    set 
    {
        _duration = value;
       DurationFormatted = _duration.HasValue ? ... : ...;
    }
}

string DurationFormatted { get; }
```

Reply:
> I want to provide a common, consistent way across all adapters to format the duration.

In this case I think it is better to create a separate class to do just that:

```
public interface IBuildDurationFormatter 
{
    string Format(long? buildDuration);
}


public sealed class BuildDurationFormatter : IBuildDurationFormatter
{
    public string Format(long? buildDuration) 
    {
           if (durationMilliseconds.HasValue)
           {
               return " (" + TimeSpan.FromMilliseconds(durationMilliseconds.Value).ToString(@"mm\:ss") + ")";
            }
            return string.Empty;
    }
}
```

Reply:
How is DurationFormatted  better than DurationText()? What is the introduced behavior you are talking about that DurationText introduces and why DurationFormatted does not introduce it?

Don't you feel that BuildDurationFormatter  is a little over engineered? It does not have any dependencies nor state - it is just a static function.